Source/SwiftLintBuiltInRules/Rules/Lint/UIImageIncludesBundleRule.swift

```diff
@@ -0,0 +1,100 @@
+        kind: .lint,
+        nonTriggeringExamples: [
+            Example("""
+            UIImage(named: "image", in: Bundle.main)
+            """),
+            Example("""
+            UIImageView(image: UIImage(named: "image", in: Bundle(for A.self)))
+            """),
+            Example("""
+            UIImage(named: "image", in: Bundle(for A.self))
+            """),
+            Example("""
+            UIImage(named: "image", in: Bundle.main, compatibleWith: aCollection)
+            """),
+            Example("""
+            UIImage(named: "image", in: Bundle.main, with: aConfiguration)
+            """),
+            Example("""
+            UIImage(systemName: "systemImage")
+            """),
+            Example("""
+            UIImage(systemName: "systemImage", compatibleWith: aCollection)
+            """),
+            Example("""
+            UIImage(systemName: "systemImage", withConfiguration: aConfiguration)
+            """),
+            Example("""
+            UIImage(systemName: "systemImage", variableValue: 0.5)
+            """),
+            Example("""
+            UIImage(systemName: "systemImage", variableValue: 0.5, configuration: aConfiguration)
+            """),
+            Example("""
+            UIImage(
+                named: "image",
+                in: Bundle.main
+            )
+            """),
+            Example("""
+            arbitraryFunctionCall("something")
+            """),
+            Example("""
+            UIImageView(image: UIImage(named: "hashtag", in: Bundle.main))
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            ↓UIImage(named: "image")
+            """),
+            Example("""
+            UIImageView(image: ↓UIImage(named: "image"))
+            """),
+            Example("""
+            ↓UIImage(named: "image", compatibleWith: aCollection)
+            """),
+            Example("""
+            ↓UIImage(named: "image", with: aConfiguration)
+            """),
+            Example("""
+            ↓UIImage(
+                named: "image"
+            )
+            """)
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension UIImageIncludesBundleRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override func visitPost(_ node: FunctionCallExprSyntax) {
+            if let identifierExpr = node.calledExpression.as(IdentifierExprSyntax.self),
+               identifierExpr.identifier.tokenKind == .identifier("UIImage"),
+               node.argumentList.containsArgument(named: "named"),
+               !node.argumentList.containsArgument(named: "in") {
+                violations.append(node.positionAfterSkippingLeadingTrivia)
+            }
+        }
+    }
+}
+
+private extension TupleExprElementListSyntax {
```

Comment:
This is a copy of the same extension in `NSLocalizedStringRequireBundleRule`. Please move it to `SwiftSyntax+SwiftLint.swift` to make it reusable.

Reply:
This pushed that file beyond 400 lines!  I tried working around this by moving any extensions on structures defined inside the generated `SyntaxCollections` (these three: `AttributeListSyntax`, `ModifierListSyntax`, and `TupleExprElementListSyntax`), but I get this error: `testSwiftLintLints(): failed - File name should match a type or extension declared in the file (if any)` This is on a file I called SwiftSyntaxCollections+SwiftLint.swift. How would you recommend getting around this?

Reply:
The `file_length` rule has meanwhile been disabled in `SwiftSyntax+SwiftLint.swift`. You don't have to bother with that anymore. I'd rather like to have these extension in one place.